test/test_smem_reuse.cpp

```diff
@@ -262,29 +262,36 @@ TEST_F(SmemReuseTest, PromoteReuse) {
 }
 
 // In this example, we promote a single tensor for re-use in a Fusion with two
-// downstream tensors that could use its memory. The first downstream tensor is
-// not re-used since it is not promoted.
+// downstream tensors that could use its memory.
 TEST_F(SmemReuseTest, PromoteReuseMultipleDownstream) {
   auto fusion = std::make_unique<Fusion>();
   FusionGuard fg(fusion.get());
 
   int64_t H = 7;
 
+  // The outer live intervals of tv0, tv2, and tv4 will be non-overlapping, but
+  // adjacent. tv0->promoteReuse() should be able to insert a sync before tv2,
+  // so that tv4 re-uses the memory from tv0, then tv2 is stacked above tv4.
```

Comment:
Why is the memory of `tv0` reused by `tv4` rather than `tv2`?

Reply:
When we clear the holding area at the end of the allocation pass, we order tv2 and tv4 by last read, so tv4 is placed below tv2. This has an effect when there are further allocations, but there are not in this case.